### PR TITLE
6058 components eodhd apis

### PR DESCRIPTION
**Pull request by luancazarine**
Triggers:
  - Stock Price Updated: Triggered when historical stock prices for a specific symbol are updated.
  - Financial Report Released: Triggered when a company releases a new financial report.

Actions:
  - Retrieve Stock Prices: Fetch historical stock prices for a given symbol and time period.
  - Get Company Financials: Obtain financial statements for a specific company.
  - Search Stock Symbol: Find stock symbols using company names or partial symbols.
  
### <samp>🤖 Generated by Copilot at f68d19c</samp>

This pull request adds a new app called `eodhd_apis` that integrates with the EOD Historical Data API to provide various components for retrieving and monitoring financial data. It also removes an unused component and updates some dependencies in the `pnpm-lock.yaml` file.

### <samp>🤖 Generated by Copilot at f68d19c</samp>

> _The `eodhd_apis` app is growing_
> _With new components for data showing_
> _It can fetch reports and prices_
> _And find symbols with advices_
> _Using the EOD Historical Data flowing_

### <samp>🤖 Generated by Copilot at f68d19c</samp>

*  Add three new action components to the eodhd_apis app that allow users to get financial statements, historical stock prices, and stock symbols using the EOD Historical Data API ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-38abf39b873a07afc80bba082cd1b931f8f26814d78112710d86c006a2aed027R1-R43), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-20fbdeb14d4b95a14193f1fce73370dacd4597bd108868f8c07e33322b060a65R1-R79), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-c3b5d829dc1c601ee1d5ab47a9580b18dd2b2c5ce902e00c3ec7d61200dd0531R1-R73))
* Add two new source components to the eodhd_apis app that emit events when a new financial report is released or a stock price is updated for a specific symbol using the EOD Historical Data API ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-265098dddcfc34107af2cbae2dfaafc2448573256e96e1985681f1c4d9f1b84fR1-R101), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-54bef3d24f763acd39566adab2022b1d8367eb3f0c341a76c30fae39759282a6R1-R84))
* Add two new files that contain constants and utility functions for the eodhd_apis app, such as sanitizing request parameters and defining possible values for some parameters ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-dcbf0fe2521b270502d5e7bc40f025ea5c8211b12079631d4136760a0878b221R1-R24), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-309a09048ac04aae7effed2c8d8274eec2f7dfe5c5ceca15e23b9e1e4061f648R1-R16))
* Add the main app file for the eodhd_apis app, which defines the app properties, methods, and authentication logic, and wraps the EOD Historical Data API endpoints ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-480bbec575ea2e27f70953944a151033cce7dade9d7e20a186edd6184da2fa62R1-R170))
* Update the package.json file for the eodhd_apis app, changing the main entry point to the app file and incrementing the version number ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-5c561051a454a5fb373e8eadfbc302f8512592af610babe5a05df07ab698691cL3-R11))
* Remove the danny_test_app component from the pnpm-lock.yaml file, which locks the dependency versions for the repository ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL885-L887))
* Add the omniconvert component to the pnpm-lock.yaml file, which is another app component that was added to the repository, and specify the dependency version for the `@pipedream/platform` package for it ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2676-R2681))
* Update the dependency versions for the `@types/node` package for various packages in the pnpm-lock.yaml file, from 18.15.11 to 17.0.45, as a result of running the pnpm install command or updating the jest packages to the latest version ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8757-R8760), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8766-R8769), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8852-R8855), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL8962-R8965), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9001-R9004), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL9081-R9084), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11439-R11442), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11446-R11449), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11474-R11477), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11499-R11502), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11506-R11509), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11512-R11515), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11560-R11563), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11566-R11569), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11674-R11677), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11685-R11688), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11702-R11705), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11723-R11726), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11740-R11743), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11746-R11749), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL11776-R11779), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17238-R17241), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17526-R17529), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17567-R17570), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17684-R17687), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17815-R17818), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17876-R17879), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL17981-R17984), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18031-R18034), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL18052-R18055), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21231-R21234), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL21251-R21254))
* Update the dependency versions for the ieee754 package for the buffer package and the string-width package for the wrap-ansi package in the pnpm-lock.yaml file, as a result of running the pnpm install command or updating the wrap-ansi package to the latest version ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL12888-R12891), [link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL24833-R24836))
* Delete the `components/eodhd_apis/.gitignore` file ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-44470c109110fb51be3a3bcafaa95b5e24c9cba4eec04704ec2755df7033344b))
* Delete the `components/eodhd_apis/app/eodhd_apis.app.ts` file ([link](https://github.com/PipedreamHQ/pipedream/pull/6095/files?diff=unified&w=0#diff-44fbbfa20626bb7fdd5e85986fda3fde7f25ff1b2aa3afc722d9f3bd555597b0))

### CODE REVIEW

File: components/eodhd_apis/actions/get-company-financials/get-company-financials.mjs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import eodhdApis from "../../eodhd_apis.app.mjs";
+
+export default {
+  key: "eodhd_apis-get-company-financials",
+  name: "Get Company Financials",
+  version: "0.0.2",
```
Comment (feyzullah) [inline components/eodhd_apis/actions/get-company-financials/get-company-financials.mjs:6]:
```suggestion
  version: "0.0.1",
```

**Review by feyzullah — APPROVED** [review]
LGTM! Please just check component version of `get-company-financials`

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[EODHD_APIs_6058_2104.pdf](https://github.com/PipedreamHQ/pipedream/files/11296381/EODHD_APIs_6058_2104.pdf)

**Comment by vunguyenhung** [comment]
Hi everyone, I can not test all of the actions and sources due to API limit. I will skip them

Test report
[EODHD_APIs_6058_1536.pdf](https://github.com/PipedreamHQ/pipedream/files/11351653/EODHD_APIs_6058_1536.pdf)

**Review by lcaresia — APPROVED** [review]

**Pull request merged**